server/src/test/java/org/jetbrains/bsp/bazel/server/sync/languages/rust/RustWorkspaceResolverTest.kt

```diff
@@ -0,0 +1,307 @@
+  }
+
+  @Test
+  fun `should return proper package and empty dependencies for a single local target`() {
+    // given
+    val features = listOf(
+        "a_feature",
+        "another_feature"
+    )
+    val rustModule = createRustModule(
+        crateId = "SampleTarget",
+        crateRoot = "/path/to/target",
+        crateFeatures = features,
+        dependencies = emptyList()
+    )
+    val module = createModule(
+        label = "/path/to/target:SampleTarget",
+        directDependencies = emptyList(),
+        rustModule = rustModule
+    )
+    val modules = listOf(module)
+
+    // when
+    val packages = resolver.rustPackages(modules)
+    val (dependencies, rawDependencies) = resolver.rustDependencies(packages, modules)
+
+    // then
+    packages.size shouldBe 1
+
+    val pkg = packages[0]
+    pkg.id shouldBe rustModule.crateRoot
+    pkg.version shouldBe rustModule.version
+    pkg.edition shouldBe rustModule.edition
+    pkg.features.map { it.name } shouldContainExactlyInAnyOrder features
+    pkg.enabledFeatures shouldContainExactlyInAnyOrder features
+
+    pkg.targets shouldNotBe null
+    pkg.targets shouldBe pkg.allTargets
+    pkg.targets.size shouldBe 1
+
+    val target = pkg.targets[0]
+    target.name shouldBe rustModule.crateId
+    target.crateRootUrl shouldBe rustModule.crateRoot
+    target.kind shouldBe rustModule.kind
+    target.edition shouldBe rustModule.edition
+    target.requiredFeatures shouldContainExactlyInAnyOrder features
+
+    dependencies shouldNotBe null
+    dependencies shouldBe emptyList()
+
+    rawDependencies shouldNotBe null
+    rawDependencies shouldBe emptyList()
+  }
+
+  @Test
+  fun `should return empty dependency list for empty package list`() {
+    // given
+    val modules = emptyList<Module>()
+    val packages = emptyList<RustPackage>()
+
+    // when
+    val (dependencies, rawDependencies) = resolver.rustDependencies(packages, modules)
+
+    // then
+    dependencies shouldNotBe null
+    dependencies shouldBe emptyList()
+
+    rawDependencies shouldNotBe null
+    rawDependencies shouldBe emptyList()
+  }
+
+  @Test
+  fun `should return proper dependency list for a target with one dependency`() {
+    // given
+    //   B
+    //   |
+    //   A
+    val rustModuleA = createRustModule(
+        crateId = "TargetA",
+        crateRoot = "/path/to/target/a",
+        dependencies = emptyList()
+    )
+    val moduleA = createModule(
+        label = "/path/to/target/a:TargetA",
+        directDependencies = emptyList(),
+        rustModule = rustModuleA
+    )
+
+    val rustModuleB = createRustModule(
+        crateId = "TargetB",
+        crateRoot = "/path/to/target/b",
+        dependencies = listOf(
+            RustDependency(
+                crateId = rustModuleA.crateId,
+                rename = rustModuleA.crateId
+            )
+        )
+    )
+    val moduleB = createModule(
+        label = "/path/to/target/b:TargetB",
```

Comment:
Labels start with `@//` or `//` before Bazel 6. Don't we have to follow those naming conventions? Underlying code could assume labels are of this form?